internal/providers/msentraid/himmelblau.go

```diff
@@ -0,0 +1,188 @@
+package msentraid
+
+//go:generate ./generate-himmelblau.sh
+
+/*
+#cgo LDFLAGS: -L${SRCDIR} -Wl,-rpath=${SRCDIR} -lhimmelblau
+#include "himmelblau.h"
+*/
+import "C"
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"unsafe"
+
+	"github.com/ubuntu/authd/log"
+	"golang.org/x/oauth2"
+)
+
+var tpm *C.BoxedDynTpm
+var authValue *C.char
+var loadableMachineKey *C.LoadableMachineKey
+var machineKey *C.MachineKey
+var brokerClientApp *C.BrokerClientApplication
+
+func init() {
+	var err *C.MSAL_ERROR
+
+	err = C.set_global_tracing_level(C.TRACE)
+	if err != nil {
+		panic(fmt.Sprintf("failed to set global tracing level: %v", C.GoString(err.msg)))
+	}
+
+	// An optional TPM Transmission Interface. If this parameter is NULL, a soft TPM is initialized.
+	var tctiName *C.char
+	err = C.tpm_init(tctiName, &tpm)
+	if err != nil {
+		panic(fmt.Sprintf("failed to initialize TPM: %v", C.GoString(err.msg)))
+	}
+
+	err = C.auth_value_generate(&authValue)
+	if err != nil {
+		panic(fmt.Sprintf("failed to generate auth value: %v", C.GoString(err.msg)))
+	}
+
+	err = C.tpm_machine_key_create(tpm, authValue, &loadableMachineKey)
+	if err != nil {
+		panic(fmt.Sprintf("failed to create machine key: %v", C.GoString(err.msg)))
+	}
+
+	err = C.tpm_machine_key_load(tpm, authValue, loadableMachineKey, &machineKey)
+	if err != nil {
+		panic(fmt.Sprintf("failed to load machine key: %v", C.GoString(err.msg)))
+	}
+}
+
+var registered = false
+
+func (p *Provider) registerDevice(ctx context.Context, token *oauth2.Token, tenantID, domain string) error {
+	if registered {
+		return nil
+	}
+
+	var err *C.MSAL_ERROR
+
+	authority := C.CString("https://login.microsoftonline.com/" + tenantID)
```

Comment:
```suggestion
	authority := C.CString("https://login.microsoftonline.com/" + tenantID)
	defer C.free(unsafe.Pointer(authority))
```

`broker_init` is not taking the ownership of it, right?